### PR TITLE
Scaffolding for charmhub charm resources command

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following starts to plan out adding charmhub queries for getting
charm resources. The code in here is old and weird. I've updated it to
allow us to perform different requests based on the charm schema.

## QA steps

Existing tests pass

```sh
$ juju bootstrap lxd test
$ juju charm-resources cs:postgresql
$ juju charm-resources mysql
ERROR charmhub charms are currently not supported
```

## Bug reference

Addresses, but doesn't complete: https://bugs.launchpad.net/juju/+bug/1913754

### CODE REVIEW

File: cmd/juju/resource/charmresources.go [hunk 1]
```diff
@@ -156,43 +191,74 @@ Where the series is not supplied, the series from your local host is used.
...
-	bakeryClient, err := c.BakeryClient()
+func resolveCharm(raw string) (*charm.URL, error) {
+	charmURL, err := charm.ParseURL(raw)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	conAPIRoot, err := c.NewControllerAPIRoot()
+
+	if charmURL.Series == "bundle" {
+		return nil, errors.Errorf("charm bundles are not supported")
```
Comment (hmlanigan) [inline cmd/juju/resource/charmresources.go:201]: Alternative: "Bundles have no resources" or along those lines.

**Review by hmlanigan — APPROVED** [review]
One wording suggestion. QA went well.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**